packages/utxo-lib/src/coinselect/coinselectUtils.ts

```diff
@@ -96,26 +98,28 @@ export function transactionBytes(inputs: Vin[], outputs: VinVout[]) {
     return Math.ceil(transactionWeight(inputs, outputs) / 4);
 }
 
-export function dustThreshold(feeRate: number, options: CoinSelectOptions) {
-    const size = transactionBytes(
-        [
-            {
-                type: options.txType,
-                script: {
-                    length: INPUT_SCRIPT_LENGTH[options.txType],
-                },
-            },
-        ],
-        [
-            {
-                script: {
-                    length: OUTPUT_SCRIPT_LENGTH[options.txType],
-                },
-            },
-        ],
-    );
-    const price = getFeeForBytes(feeRate, size);
-    return Math.max(options.dustThreshold, price);
+// https://github.com/bitcoin/bitcoin/blob/be443328037162f265cc85d05b1a7665b5f104d2/src/policy/policy.cpp#L28-L41
+// Absolute minimum accepted dustRelayFeeRate is set to 3 sat/byte (3000 sat/kB) in bitcoin-core.
+// To make potential output spendable in the near future it could be slightly higher basing on longTermFeeRate (if provided)
+// Minimum dust amount is the greater value from:
+// - provided constant `dustThreshold` (typically 546 sat. with few exceptions like DOGE)
+// - calculated from inputSize multiplied by dustRelayFeeRate
+export function getDustAmount(
+    feeRate: number,
+    { txType, longTermFeeRate, dustThreshold }: CoinSelectOptions,
+) {
+    const inputSize = inputBytes({
+        type: txType,
+        script: {
+            length: INPUT_SCRIPT_LENGTH[txType],
+        },
+    });
+    // use current feeRate if it is lower than longTermFeeRate
+    const longTermFee = longTermFeeRate ? Math.min(feeRate, longTermFeeRate) : 0;
```

Comment:
well, if you set it by its own it will be **always** picked by the next line and dustRelayFeeRate will equal feeRate (this is what im trying to fix here)

Reply:
in this line we are deciding what is longTermFee, and it should be set only if longTermFeeRate is set.
but if its set and it's higher that requested feeRate then it should be lower :) hope that make sense now

Reply:
> it will be always picked

Not always, only when it's greater than 3, but I get the point.

>  if its set and it's higher that requested feeRate then it should be lower

Ok so we don't want to approximate longTermFee from feeRate only. It's just that when we have both and feeRate is lower, it means that the longTermFeeRate approximation was wrong and we adjust it, right?

Reply:
right :) explained IRL